### ISSUE TITLE
Problem with 'select' type

### ISSUE THREAD

**Issue by theochr72** [sqlpage/SQLPage#757]
I want to create a dropdown menu in a form, and I want to set a default value and then sort the options alphabetically. However, I feel like setting a default value is impossible.

Do you have any other solution or suggestion to assign a default value?

When I query my database, "Europe/Paris" appears first, but not on the SQL page.

Here is my SQL query:

```sql
SELECT
    'Timezone' AS label,
    'timezone' AS name,
    'select' AS type,
    '' AS description,
    FALSE AS create_new,
    'Europe/Paris' AS placeholder,
    TRUE AS searchable,
    TRUE AS required,
    json_agg(json_build_object('label', name, 'value', name)) AS options
FROM (
         SELECT DISTINCT
             name,
             CASE
                 WHEN name = 'Europe/Paris' THEN 0
                 ELSE 1
                 END AS paris_first
         FROM
             timezone
         ORDER BY
             paris_first,
             name
     ) AS sorted
GROUP BY
    label,
    type,
    description,
    create_new,
    placeholder,
    searchable,
```

**Comment by lovasoa** [comment]
Hello and welcome to SQLPage !

If what you need is just to set a default selected option, you can use the `selected` key inside the `options` property of [the form component](https://sql-page.com/component.sql?component=form):

```sql
SELECT
    'Timezone' AS label,
    'timezone' AS name,
    'select' AS type,
    TRUE AS searchable,
    json_agg(json_build_object(
        'label', name,
        'value', name,
        'selected', name = 'Europe/Paris'
    ) order by name) AS options
FROM timezone;
``` 

![image](https://github.com/user-attachments/assets/54c6dcd5-ec7d-4e38-8c30-d8c807bb7bf9)

The postgres documentation has the following to say about [ordering the results of json_agg](https://www.postgresql.org/docs/9.5/functions-aggregate.html) (emphasis mine): 
> aggregate functions produce meaningfully different result values depending on the order of the input values. This ordering is unspecified by default, but can be controlled by writing an ORDER BY clause within the aggregate call, as shown in [Section 4.2.7](https://www.postgresql.org/docs/9.5/sql-expressions.html#SYNTAX-AGGREGATES). Alternatively, supplying the input values from a sorted subquery **will usually work**.

---

If you really need to change the order of options, the query becomes

```sql
SELECT
    json_agg(
      json_build_object(
        'label', name,
        'value', name,
        'selected', name = 'Europe/Paris'
      )
      order by 
             case name when 'Europe/Paris' then 0 ELSE 1 END,
             name
    ) AS options
FROM timezone;
```

But I'm not sure this is very useful :stuck_out_tongue:

**Comment by theochr72** [comment]
Indeed, that was exactly what I was looking for, thank you for your help. I see that you're using https://editor.datapage.app/; I just discovered it, and it looks pretty cool.
